core/controlplane/config/templates/stack-template.json

```diff
@@ -257,15 +242,37 @@
                   "Resource": "*",
                   "Effect": "Allow"
                 }
+          ]
+        }
+      }
+    },
+    "IAMRoleController": {
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Statement": [
+            {
+              "Action": [
+                "sts:AssumeRole"
               ],
-              "Version": "2012-10-17"
-            },
-            "PolicyName": "root"
-          }
-        ]
+              "Effect": "Allow",
+              "Principal": {
+                "Service": [
+                  "ec2.{{.Region.PublicDomainName}}"
+                ]
+              }
+            }
+          ],
+          "Version": "2012-10-17"
+        },
+        "Path": "/",
+        {{if .Controller.IAMConfig.ManagedIamRoleName }}
+        "RoleName":  {"Fn::Join": ["",[{"Ref": "AWS::Region"},"-","{{.Controller.IAMConfig.ManagedIamRoleName}}"]]},
+        {{end}}
+        "ManagedPolicyArns": [ {"Ref": "IAMManagedPolicyController"} ]
```

Comment:
Sounds great. I'm amost convinced to your idea!

Nit but:
* Would you mind renaming `iamRoleName` to just `name` so that the whole path would look like `iam.role.name` rather than `iam.role.iamRoleName`, which results in less duplication?
* Which of them do you think clearer, `iam.instanceProfile.role` or `iam.role`? My motivation to nest `role` under `instanceProfile` is based on the fact that the role is associated to the instance profile. However I'd really like to hear your idea.